keiko-sql/src/main/kotlin/com/netflix/spinnaker/q/sql/SqlQueue.kt

```diff
@@ -201,7 +200,7 @@ class SqlQueue(
           .intoResultSet()
       }
 
-      while (!found && rs.next()) {
+      while (!found && rs.row < batchSize && rs.next()) {
```

Comment:
The underlying problem here is a little subtle. The bug appears to be the result of the [ResultSet.next()](https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/ResultSet.html#next()) overshooting the cursor at the end of searching a batch. 

The resulting [ResultSet.row()](https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/ResultSet.html#getRow()) check [below](https://github.com/spinnaker/orca/blob/4b9b856bfe2720766f87c52c8d4a388eba7a8a1b/keiko-sql/src/main/kotlin/com/netflix/spinnaker/q/sql/SqlQueue.kt#L215) never equals the expected `batchSize` and  so no further batches are queried when they should.